examples/pokemon-service-tls/tests/common/mod.rs

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use std::{fs::File, io::BufReader, process::Command, time::Duration};
+
+use assert_cmd::prelude::*;
+use aws_smithy_client::{
+    erase::{DynConnector, DynMiddleware},
+    hyper_ext::Adapter,
+};
+use tokio::time::sleep;
+
+use pokemon_service_client::{Builder, Client, Config};
+use pokemon_service_common::{rewrite_base_url, ChildDrop};
+use pokemon_service_tls::{DEFAULT_DOMAIN, DEFAULT_PORT, DEFAULT_TEST_CERT};
+
+pub async fn run_server() -> ChildDrop {
+    let child = Command::cargo_bin("pokemon-service-tls")
+        .unwrap()
+        .spawn()
+        .unwrap();
+
+    sleep(Duration::from_millis(500)).await;
+
+    ChildDrop(child)
```

Comment:
When I have done it, I did not know about `kill_on_drop`. I think we can just make the change you are suggesting.

Reply:
As far as I can tell [`kill_on_drop`](https://docs.rs/tokio/latest/tokio/process/struct.Command.html#method.kill_on_drop) is a `tokio::process::Command` only method.

We could switch over to `tokio::process::Command` except we'd lose the ability to use `Command::cargo_bin`, from the [`assert_cmd`](https://github.com/assert-rs/assert_cmd) crate - which unfortunately doesn't implement the [extension method](https://docs.rs/assert_cmd/latest/assert_cmd/cargo/trait.CommandCargoExt.html) on the `tokio` type.

It seems we only use `assert_cmd` for this one use case and the implementation of `cargo_bin` is quite simple - I can inline this and switch us to `tokio::process::Command`, but it might be more hassle than it's worth?

Reply:
After your analysis, I think we should keep what we have.

Reply:
Fine to leave it as is for me as well.